106-auto-rebalance-on-imbalanced-clusters.md

```diff
@@ -0,0 +1,534 @@
+# ...
+```
+
+The advantages of using a single ConfigMap for every anomaly are:
+1. Single place to put in all the anomalies
+2. Even when dealing multiple Kafka clusters, the ConfigMap created will be equal to number of clusters deployed
+
+This generated config map will contain the list of goal violation anomalies in the section named `anomaly_list`. The ConfigMap will also store the timestamps for when the last anomaly detection was run and when the rebalance is finished.
+Whenever a rebalance is finished, the `anomalyList` would be set to empty by the operator. 
+
+The `last_anomaly_detection_timestamp` corresponds to when the last anomaly detection was run and will be updated everytime a new anomaly detection is run and the `rebalance_finished_timestamp` refers to the time when the running rebalance is finished and will be updated everytime when the running rebalance is finished.
+
+The created ConfigMap will persist when self-healing is enabled and will be deleted if the user decides to disable self-healing. This is because the ConfigMap is going to help us track the timestamps during the complete process.
+
+#### What happens if same anomaly is detected while the auto-rebalance is happening
+
+The `last_anomaly_detection_timestamp` was before `rebalance_finished_timestamp` timestamps will allow us to tackle cases where the rebalance was happening but anomaly detection also took place in that time period and detects the same anomaly which is being fixed by the rebalance.
+Once the rebalance in finished, we can compare the `last_anomaly_detection_timestamp` and `rebalance_finished_timestamp` to compare whether the detection took place before `rebalance_finished_timestamp` or not.
+If the `last_anomaly_detection_timestamp` was before `rebalance_finished_timestamp`, we can then ignore all the anomalies.
+We don't have to worry about the loss of any other new anomalies which was detected since Cruise Control will detect them again if they persist.
+
+#### Using other notifiers when using `imbalance` mode
+
+Users cannot configure the notifier if they are utilising the auto-rebalance on imbalanced cluster.
+This is because the operator is using our custom notifier for getting alerts about goal violations. 
+If the users try to override the notifier while the `imbalance` mode is enabled, then the operator would throw warnings in the status field of the Kafka CR and ignore the changes made by the user
+
+```yaml
+status:
+  autoRebalance:
+    state: Idle
+    lastTransitionTime: "2025-09-22T16:04:00Z"
+    modes:
+      - mode: imbalance
+  conditions:
+    - type: Warning
+      status: "True"
+      reason: UnsupportedOperationException
+      message: |
+        Notifier cannot be overridden since `imbalance` mode is enabled in the `auto-rebalance` configuration
+```
+
+If the users really want to have their own way of dealing with the imbalanced clusters then they can disable auto-rebalance in `imbalance` mode and use their own notifier. 
```

Comment:
It may be possible for our notifier to forward notifications to other configured notifiers, but this would a future enhancement if users want it.